backend/prisma/schema.prisma

```diff
@@ -0,0 +1,104 @@
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+model User {
+  id                Int                @id @default(autoincrement())
+  username          String             @unique
+  email             String             @unique
+  password          String
+  workoutTemplates  WorkoutTemplate[]
+  workoutSessions   WorkoutSession[]
+  following  Follow[]           @relation("UserFollowing")
```

Comment:
Question: Does setting a relation in prisma automatically set an index on the table? Eg. if I want to query all the users who user1 follows, is it O(1) or O(N).

Reply:
I don't know about this, however, I don't think it sets an index so your example would probably be an O(N) operation.

Reply:
sounds good. as a stretch goal if we get to it, or if we notice the app is loading slowly, I would suggest us adding an index to this table to make it scale a bit better